Expensify/App
# Improve performance of all screens using OptionsList

### Details
We were rendering a very large initial set of data for anything using the `OptionsList`. I can't remember why it was set up like that, but removing it doesn't seem to have any negative effects so far.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/155388

### Tests
1. Click on the magnifying glass icon
2. Verify that the Search screen loads quickly

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
